### ISSUE TITLE
Disable indexing and watching of files in special directories

### ISSUE THREAD

**Issue by dlsniper** [bazelbuild/intellij#2032]
Hello, I'm forwarding the following piece of feedback for the plugin:

> - by default bazel creates several "convenient symlinks" to the bazel-bin/bazel-out directory within the workspace, containing several million files. Opening a project with those folders nearly crashes Goland. Luckily it's possible to disable those symlinks in the bazelrc file.

As a proposed solution, it would be better if the user doesn't need to disable the symlink feature in the bazelrc but have the plugin automatically ignore the `bazel-bin` and `bazel-out` from the indexing and watching for changes features.

[Original comment with the feedback](https://www.reddit.com/r/golang/comments/i747w4/take_part_in_the_go_development_research_from/g10uh0r/).

Thank you.

**Comment by rdesgroppes** [comment]
On my maching, `strace`ing the `fsnotifier` process indeed shows _millions_ of watches on Bazel cache entries looking like:
```
inotify_add_watch(3, "/home/username/.cache/bazel/_bazel_regis/unique_id/execroot/...
inotify_add_watch(3, "/home/username/.cache/bazel/_bazel_regis/unique_id/external/...
inotify_add_watch(3, "/home/username/.cache/bazel/_bazel_regis/install/another_unique_id/...
```

**Comment by rdesgroppes** [comment]
A workaround consists in a filtering wrapper script:
```sh
#!/usr/bin/env sh
exec grep --line-buffered -v -F /_bazel_ | /path/to/jetbrains/fsnotifier" "$@"
```

**Comment by sgowroji** [comment]
@rdesgroppes Thanks for sharing the [workaround](https://github.com/bazelbuild/intellij/issues/2032#issuecomment-1256291900).  @dlsniper Feel free to reach if you still have any further queries on the same.

**Comment by dlsniper** [comment]
@sgowroji while the workaround is good, I don't think this issue is closed.
I don't expect many/all people using the plugin to come to this thread and apply the workaround.
The plugin should really take care of this on its own and handle this case.

**Comment by rdesgroppes** [comment]
Right, such a workaround had only be crafted to buy some time. Thanks for having reopened the issue.

**Comment by jschaf** [comment]
Still relevant: @bazelbuild/triage isn't a GitHub team so not sure who to reach out to.

**Comment by agribov** [comment]
Commenting to chime in - This is still an issue and was causing crashes of my IDE - Workaround script above seems to work for me, but this would definitely be good to build into the Bazel plugin.

**Comment by agribov** [comment]
@mai93 I noticed you put the P3 label for "not considering working on this" - Could you justify that or consider updating this? This caused significant IDE issues and crashes for me until I found this workaround and rendered the plugin basically unusable for a large repository, and seems like it's a pretty simple fix.
